syntax = "proto2";

package openxtf;

import 'xtf.proto';


message XTFFrontendEvent {
  required int32 cell_number = 1;
  optional string serial_number = 2;
  optional bool popup_result = 3;  // True for YES or OKAY, False for NO.
}

message XTFStationResponse {
  optional string station_name = 1;
  optional string framework_version = 2;
  optional TestInfo test_info = 3;
  repeated XTFCell cells = 4;
}

message XTFCell {
  required int32 cell_number = 1;
  optional TestRun test_run = 2;
  optional Popup popup = 3;
}

message Popup {
  enum PopupStyle {
    CONFIRM = 1;   // OKAY button
    YESNO = 2;     // YESNO
  }

  optional string title = 1;
  optional string content = 2;
  optional PopupStyle style = 3;
}
